solr/core/src/java/org/apache/solr/search/Grouping.java

```diff
@@ -461,7 +461,16 @@ private void searchWithTimeLimiter(final Query filterQuery, Collector collector)
       searcher.search(QueryUtils.combineQueryAndFilter(query, filterQuery), collector);
     } catch (TimeLimitingCollector.TimeExceededException
         | ExitableDirectoryReader.ExitingReaderException x) {
-      log.warn("Query: {}; ", query, x);
+      if (log.isWarnEnabled()) {
+        String queryToLog = query.toString();
+        if (queryToLog.length() > 1000) {
+          // INFO log long queries separately
+          log.info("Query: {}; ", query);
```

Comment:
did you intend to log an `info` line inside the `.isWarnEnabled` block ?

Reply:
yes, that was intended but actually looking at all this again now, the info/warn split and separation of query/exception logging could potentially be done always i.e. not only for long queries. pushed a commit for that alternative.